### PR TITLE
ut for reverse replication shadow tables

### PR DESCRIPTION
**Pull request by aksharauke**

### CODE REVIEW

File: v2/spanner-to-sourcedb/src/main/java/com/google/cloud/teleport/v2/templates/utils/ShadowTableCreator.java [hunk 1]
```diff
@@ -47,6 +47,8 @@ public class ShadowTableCreator {
   private final SpannerConfig spannerConfig;
   private final SpannerConfig metadataConfig;
   private String shadowTablePrefix;
+  private Ddl informationSchemaOfMainDb;
```
Comment (shreyakhajanchi) [inline v2/spanner-to-sourcedb/src/main/java/com/google/cloud/teleport/v2/templates/utils/ShadowTableCreator.java:50]: nit: mainDb sounds a bit odd , will it be better to use primaryDb?

File: v2/spanner-to-sourcedb/src/test/java/com/google/cloud/teleport/v2/templates/utils/ShadowTableCreatorTest.java [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+import java.util.List;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public final class ShadowTableCreatorTest {
+
+  @Test
+  public void behaviorBeingTested_expectedResult() {
```
Comment (shreyakhajanchi) [inline v2/spanner-to-sourcedb/src/test/java/com/google/cloud/teleport/v2/templates/utils/ShadowTableCreatorTest.java:34]: The name of the test method is very confusing and gives no idea of what is being tested.

File: v2/spanner-to-sourcedb/src/main/java/com/google/cloud/teleport/v2/templates/utils/ShadowTableCreator.java [hunk 3]
```diff
@@ -104,15 +135,15 @@ public void createShadowTablesInSpanner() {
...
     Table.Builder shadowTableBuilder = Table.builder(dialect);
     String shadowTableName = shadowTablePrefix + dataTableName;
     shadowTableBuilder.name(shadowTableName);
 
     // Add key columns from the data table to the shadow table builder.
-    Table dataTable = informationSchema.table(dataTableName);
+    Table dataTable = informationSchemaOfMainDb.table(dataTableName);
     Set<String> primaryKeyColNames =
         dataTable.primaryKeys().stream().map(k -> k.name()).collect(Collectors.toSet());
     List<Column> primaryKeyCols =
```
Comment (shreyakhajanchi) [inline v2/spanner-to-sourcedb/src/main/java/com/google/cloud/teleport/v2/templates/utils/ShadowTableCreator.java:149]: I think the primary key builder logic is wrong, it is
`shadowTableBuilder.primaryKey().desc(keyColumn.name()).end()`, whereas it should first add all the columns and at last add `end()`

File: v2/spanner-to-sourcedb/src/test/java/com/google/cloud/teleport/v2/templates/utils/ShadowTableCreatorTest.java [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+    Ddl ddl =
+        Ddl.builder()
+            .createTable("table1")
+            .column("id")
+            .int64()
+            .endColumn()
+            .column("update_ts")
+            .timestamp()
+            .endColumn()
+            .primaryKey()
```
Comment (shreyakhajanchi) [inline v2/spanner-to-sourcedb/src/test/java/com/google/cloud/teleport/v2/templates/utils/ShadowTableCreatorTest.java:62]: can you please add one more column to the primary key to check the behaviour for multiple PK columns

File: v2/spanner-to-sourcedb/src/main/java/com/google/cloud/teleport/v2/templates/utils/ShadowTableCreator.java (hunk 1 above)
Comment (aksharauke) [reply v2/spanner-to-sourcedb/src/main/java/com/google/cloud/teleport/v2/templates/utils/ShadowTableCreator.java]: Done

File: v2/spanner-to-sourcedb/src/test/java/com/google/cloud/teleport/v2/templates/utils/ShadowTableCreatorTest.java (hunk 2 above)
Comment (aksharauke) [reply v2/spanner-to-sourcedb/src/test/java/com/google/cloud/teleport/v2/templates/utils/ShadowTableCreatorTest.java]: Done

File: v2/spanner-to-sourcedb/src/main/java/com/google/cloud/teleport/v2/templates/utils/ShadowTableCreator.java (hunk 3 above)
Comment (aksharauke) [reply v2/spanner-to-sourcedb/src/main/java/com/google/cloud/teleport/v2/templates/utils/ShadowTableCreator.java:149]: The end is just for that primary key, this is being called in loop so this is okay. I have also tested this manually. Also the logic is similar in the forward migration template.

File: v2/spanner-to-sourcedb/src/test/java/com/google/cloud/teleport/v2/templates/utils/ShadowTableCreatorTest.java (hunk 4 above)
Comment (aksharauke) [reply v2/spanner-to-sourcedb/src/test/java/com/google/cloud/teleport/v2/templates/utils/ShadowTableCreatorTest.java:78]: Done

**Review by shreyakhajanchi — APPROVED** [review]
LGTM

**Pull request merged**